BinaryEncoding.md

```diff
@@ -119,11 +119,15 @@ The type section declares all function signatures that will be used in the modul
 | entries | `type_entry*` | repeated type entries as described below |
 
 #### Type entry
-| Field | Type | Description |
+| Field | Type/Value | Description |
 | ----- |  ----- | ----- |
+| constructor | `0x05` | the function type constructor |
```

Comment:
Let me check my understanding of this interpretation of "constructor":
 * we're viewing `i32` and the other local types as having nullary constructors that construct the local type
 * `0x5`, the function constructor, constructs a function type given parameters/returns
 * in the local type declaration of a function body, we have a list of (lists of) locals that declare their type via integer index.  In a future of richer (GC) types where we had an safe/opaque function-pointer type, the way you declare a local of func-ptr type would be to first emit the index of the function constructor, `0x5` which told the decoder to decode a second index which was an index into this types section array.  At a higher level, you're saying first which constructor and then providing the arguments to the constructor.
 * If we had a "struct type" constructor, it could be, say, `0x6`, and the same scheme above would apply and the validation rules would require that the index following a local type `0x5`/`0x6` had to match on the `constructor` field

If that's all right, then that seems great.  Maybe it would be good to update GC.md to include this plan of record.